syntax = "proto3";

package pb_viewer;

message MCUDeviceInfo{
    string name = 2;
    float ratio_of_tick_to_ns = 3;
}

enum IMUAccScale{
    G_2 = 0;
    G_4 = 1;
    G_8 = 2;
    G_16 = 3;
}

enum IMUGyroScale{
    DPS_125 = 0;
    DPS_250 = 1;
    DPS_500 = 2;
    DPS_1000 = 3;
    DPS_2000 = 4;
}

message IMUDeviceInfo{
    string name = 2;
    uint32 device_id = 3;
    float frequency = 4;
    IMUAccScale acc_scale = 5;
    IMUGyroScale gyro_scale = 6;
}

message IMUAccGyro{
    uint32 device_id = 2;
    uint32 tick = 3;

    float acc_x = 4;
    float acc_y = 5;
    float acc_z = 6;

    float gyro_x = 7;
    float gyro_y = 8;
    float gyro_z = 9;
}

message IMUTemperature{
    uint32 device_id = 2;
    uint32 tick = 3;

    float temperature = 4;
}

message IMUMagnetometer{
    uint32 device_id = 2;
    uint32 tick = 3;

    float hx = 4;
    float hy = 5;
    float hz = 6;
}

message Command{

}

message Message{
    uint32 sequence_id = 1;
	oneof data{
		MCUDeviceInfo mcu_device_info = 2;
		IMUDeviceInfo imu_device_info = 3;
		IMUAccGyro imu_acc_gyro = 4;
		IMUTemperature imu_temperature = 5;
		IMUMagnetometer imu_magnetometer = 6;
	}

	
}
